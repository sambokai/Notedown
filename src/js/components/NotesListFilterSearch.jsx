import React from 'react';
import PropTypes from 'prop-types';


function NotesListFilterSearch(props) {
  return (
    <div className="form-row mb-1">
      <div className="col">
        <input
          type="text"
          className="form-control"
          placeholder="Search"
          onChange={e => props.onUpdateSearchQuery(e.target.value)}
        />
      </div>
    </div>
  );
}

NotesListFilterSearch.propTypes = {
  onUpdateSearchQuery: PropTypes.func.isRequired,
};

export default NotesListFilterSearch;
