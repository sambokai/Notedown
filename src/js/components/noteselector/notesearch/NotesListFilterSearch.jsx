import React from 'react';
import PropTypes from 'prop-types';


function NotesListFilterSearch(props) {
  return (
    <div className="form-row pb-1">
      <div className="col">
        <input
          id="note-search-input"
          type="text"
          className="form-control"
          placeholder="Search"
          onChange={e => props.onUpdateSearchQuery(e.target.value)}
        />
      </div>
    </div>
  );
}

NotesListFilterSearch.propTypes = {
  onUpdateSearchQuery: PropTypes.func.isRequired,
};

export default NotesListFilterSearch;
