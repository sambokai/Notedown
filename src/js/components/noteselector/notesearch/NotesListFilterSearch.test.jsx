import React from 'react';

import { mount, render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';

import NotesListFilterSearch from './NotesListFilterSearch';

const onUpdateSearchQueryMock = jest.fn();

describe('<NotesListFilterSearch/>', () => {
  it('should render correctly', () => {
    const wrapper = render(<NotesListFilterSearch onUpdateSearchQuery={onUpdateSearchQueryMock} />);

    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it("should call onUpdateSearchQuery callback upon writing text, with value of text-input field as it's parameter", () => {
    const wrapper = mount(<NotesListFilterSearch onUpdateSearchQuery={onUpdateSearchQueryMock} />);
    const textSearchInput = wrapper.find('input');
    const searchQuery = 'Green Bell Peppers';

    textSearchInput.instance().value = searchQuery;
    textSearchInput.simulate('change');

    expect(onUpdateSearchQueryMock.mock.calls).toHaveLength(1);
    expect(onUpdateSearchQueryMock.mock.calls[0][0]).toBe(searchQuery);
  });
});
