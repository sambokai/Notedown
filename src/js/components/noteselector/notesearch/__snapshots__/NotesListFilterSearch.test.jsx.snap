// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotesListFilterSearch/> should render correctly 1`] = `
<div
  class="form-row pb-1"
>
  <div
    class="col"
  >
    <input
      class="form-control"
      id="note-search-input"
      placeholder="Search"
      type="text"
    />
  </div>
</div>
`;
